File: bench/data-structures-bst/2.cpp
#include <iostream>
#include <ctime>
#include <cstdlib>
#include "cpp-btree/btree_set.h"
#include <chrono>

using namespace std;

int main(int argc, char const *argv[])
{
	srand(time(NULL));

	btree::btree_set<int> btree;

	int S = 1000000;
	int n = 1000000;
	if (argc >= 2)
	{
		S = atoi(argv[1]);

		if (argc == 3)
		{
			n = atoi(argv[2]);
		}
	}

	for (int i = 0; i < S; i++)
	{
		btree.insert(rand());
	}

	auto start = chrono::steady_clock::now();

	for (int i = 0; i < n; i++)
	{
		btree.erase(rand());
	}

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}

File: bench/data-structures-bst/2.go
package main

import (
	"math/rand"
	"os"
	"strconv"
	"time"

	"github.com/google/btree"
)

func main() {
	rand.Seed(time.Now().UnixNano())

	var S = 1000000
	var n = 1000000
	var args = os.Args[1:]
	if len(args) >= 1 {
		S, _ = strconv.Atoi(args[0])
		if len(args) >= 2 {
			n, _ = strconv.Atoi(args[1])
		}
	}

	var tr = btree.New(2)

	for i := 0; i < S; i++ {
		tr.ReplaceOrInsert(btree.Int(rand.Int()))
	}

	for i := 0; i < n; i++ {
		tr.Delete(btree.Int(rand.Int()))
	}
}
